Frameworks/CoreGraphics/CGContextCairo.mm

```diff
@@ -2013,6 +2014,16 @@
     CGAffineTransform transform = CGAffineTransformConcat(curState->curTransform, curState->curTextMatrix);
     // Perform translation required to handle scaling.
     CGFloat verticalScalingFactor = sqrt((transform.b * transform.b) + (transform.d * transform.d));
+    // TODO::
+    // Issue #1057 - CG today talks in Pixels but DirectWrite and D2D only understand DIPs. Because CG talks in Pixels,
+    // curState->curTransform contains both the device scaling factor and any user scale applied to it causing the below D2D render target
+    // transform calculation to be way off on devices that have scaling. Workaround for now is to divide the verticalScalingFactor and
+    // height here with the device scaling factor so they get converted to DIPs and then apply the transform to the D2D render target.
+    // Though this workaround is sufficient to get text rendered on the user screen on all Windows form factor devices, this will not work
+    // when text is rendered on a PDF or printer.
+    float hostDisplayScale = static_cast<float>([[WGDDisplayInformation getForCurrentView] resolutionScale] / 100.0f);
```

Comment:
I was about to update the CR for that when my VS stopped working for me :-|

---
In reply to: [81259574](https://github.com/Microsoft/WinObjC/pull/1058#discussion_r81259574) [](ancestors = 81259574)